src/Bicep.Core/Semantics/ParamFileSymbol.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Extensions;
+using Bicep.Core.Syntax;
+
+namespace Bicep.Core.Semantics
+{
+    public class ParamFileSymbol : Symbol
+    {
+        private readonly ILookup<string, BindableSymbol> symbolsByName;
+        private readonly Lazy<IEnumerable<ErrorDiagnostic>> lazyDiagnostics;
+
+        public ParamFileSymbol(ImmutableArray<ParameterAssignmentSymbol> parameterAssignmentSymbols,
+            string name,
+            ProgramSyntax syntax,
+            IEnumerable<BindableSymbol> symbols,
+            Uri fileUri)
+            : base(name)
+        {
+            this.lazyDiagnostics = new(() => FindDuplicateNamedSymbols(parameterAssignmentSymbols)
+                .Select(symbol => DiagnosticBuilder.ForPosition(symbol.NameSyntax).ParameterMultipleAssignments(symbol.Name)));
+
+            this.Syntax = syntax;
+            FileUri = fileUri;
+            this.ParameterAssignmentSymbols = parameterAssignmentSymbols;
+            this.symbolsByName = this.Symbols.ToLookup(symbol => symbol.Name, LanguageConstants.IdentifierComparer);
+        }
+
+        public override IEnumerable<Symbol> Descendants =>
+            this.ParameterAssignmentSymbols;
+
+        public override SymbolKind Kind => SymbolKind.File;
+
+        public ProgramSyntax Syntax { get; }
+
+        public ImmutableArray<ParameterAssignmentSymbol> ParameterAssignmentSymbols { get; }
+
+        public Uri FileUri { get; }
+
+        /// <summary>
+        /// Returns all the top-level declaration symbols.
+        /// </summary>
+        public IEnumerable<BindableSymbol> Symbols => this.Descendants.OfType<BindableSymbol>();
```

Comment:
This may not be needed since `ParamFileSymbol` can only contains `ParameterAssignmentSymbol`.